pkg/alertmanager/statefulset.go

```diff
@@ -270,6 +270,16 @@ func makeStatefulSetSpec(a *monitoringv1.Alertmanager, config Config, tlsAssetSe
 	}
 	amArgs = append(amArgs, fmt.Sprintf("--web.route-prefix=%v", webRoutePrefix))
 
+	web := a.Spec.Web
+	if version.GTE(semver.MustParse("0.17.0")) && web != nil && web.GetConcurrency != nil {
+		amArgs = append(amArgs, fmt.Sprintf("--web.get-concurrency=%d", *web.GetConcurrency))
+	}
+
+	if version.GTE(semver.MustParse("0.17.0")) && web != nil && web.Timeout != nil {
+		amArgs = append(amArgs, fmt.Sprintf("--web.timeout=%d", *web.Timeout))
+
```

Comment:
I am not sure if that helps since users would rarely read the operator logs if things work. I think the right way to do this would be in the admission webhook controller to validate the fields and reject it.

Reply:
Perhaps we should take this discussion about logging separately from this PR since if we are to do this, I think we should be consistent across all fields and not just the `web.*`